tests/components/watttime/test_config_flow.py

```diff
@@ -22,201 +23,153 @@
-    hass: HomeAssistant, config_auth, config_location_type, setup_watttime
-) -> None:
-    """Test showing the form to input custom latitude/longitude."""
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
-    )
+    # Test that we can recover from login errors:
     result = await hass.config_entries.flow.async_configure(
         result["flow_id"], user_input=config_auth
     )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "location"
+
     result = await hass.config_entries.flow.async_configure(
-        result["flow_id"], user_input=config_location_type
+        result["flow_id"], user_input={CONF_LOCATION_TYPE: LOCATION_TYPE_COORDINATES}
     )
-    result = await hass.config_entries.flow.async_configure(result["flow_id"])
-    assert result["type"] == FlowResultType.FORM
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
     assert result["step_id"] == "coordinates"
-    assert result["errors"] is None
 
-
-async def test_show_form_user(hass: HomeAssistant) -> None:
-    """Test showing the form to select the authentication type."""
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
-    )
-    assert result["type"] == FlowResultType.FORM
-    assert result["step_id"] == "user"
-    assert result["errors"] is None
-
-
-async def test_step_reauth(
-    hass: HomeAssistant, config_auth, config_coordinates, config_entry, setup_watttime
-) -> None:
-    """Test a full reauth flow."""
-    with patch(
-        "homeassistant.components.watttime.async_setup_entry",
-        return_value=True,
+    # Test errors that can arise when selecting a location:
+    with patch.object(
+        client.emissions, "async_get_grid_region", get_grid_region_response
     ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN,
-            context={"source": config_entries.SOURCE_REAUTH},
-            data={
-                **config_auth,
-                **config_coordinates,
-            },
-        )
         result = await hass.config_entries.flow.async_configure(
-            result["flow_id"],
-            user_input={CONF_PASSWORD: "password"},
+            result["flow_id"], user_input=config_coordinates
         )
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "coordinates"
+        assert result["errors"] == get_grid_region_errors
 
-    assert result["type"] == FlowResultType.ABORT
-    assert result["reason"] == "reauth_successful"
-    assert len(hass.config_entries.async_entries()) == 1
+    # Test that we can recover from location errors:
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=config_coordinates
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
```

Comment:
Config flow tests should patch `async_setup_entry`, preferably with a `mock_setup_entry` fixture in `conftest.py` and coresponding `pytestmark`
See #88484